diff --git a/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/Private/ScriptGeneratorPlugin.cpp b/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/Private/ScriptGeneratorPlugin.cpp
index 46f6936..a13a5e7 100644
--- a/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/Private/ScriptGeneratorPlugin.cpp
+++ b/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/Private/ScriptGeneratorPlugin.cpp
@@ -4,6 +4,7 @@
 #include "ScriptCodeGeneratorBase.h"
 #include "GenericScriptCodeGenerator.h"
 #include "LuaScriptCodeGenerator.h"
+#include "KlawrCodeGenerator.h"
 
 DEFINE_LOG_CATEGORY(LogScriptGenerator);
 
@@ -42,6 +43,9 @@ void FScriptGeneratorPlugin::Initialize(const FString& RootLocalPath, const FStr
 #if WITH_LUA
 	UE_LOG(LogScriptGenerator, Log, TEXT("Using Lua Script Generator."));
 	CodeGenerator = new FLuaScriptCodeGenerator(RootLocalPath, RootBuildPath, OutputDirectory, IncludeBase);
+#elif WITH_KLAWR
+	UE_LOG(LogScriptGenerator, Log, TEXT("Using Klawr Script Generator."));
+	CodeGenerator = new FKlawrCodeGenerator(RootLocalPath, RootBuildPath, OutputDirectory, IncludeBase);
 #else
 	UE_LOG(LogScriptGenerator, Log, TEXT("Using Generic Script Generator."));
 	CodeGenerator = new FGenericScriptCodeGenerator(RootLocalPath, RootBuildPath, OutputDirectory, IncludeBase);
diff --git a/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/ScriptGeneratorPlugin.Build.cs b/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/ScriptGeneratorPlugin.Build.cs
index 874a7bb..163ac5d 100644
--- a/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/ScriptGeneratorPlugin.Build.cs
+++ b/Engine/Plugins/Script/ScriptGeneratorPlugin/Source/ScriptGeneratorPlugin/ScriptGeneratorPlugin.Build.cs
@@ -52,6 +52,12 @@ namespace UnrealBuildTool.Rules
 				{
 					Definitions.Add("WITH_LUA=1");
 				}
+
+                var KlawrPath = Path.Combine(UEBuildConfiguration.UEThirdPartySourceDirectory, "Klawr");
+                if (Directory.Exists(KlawrPath))
+                {
+                    Definitions.Add("WITH_KLAWR=1");
+                }
 			}
 		}
 	}
